src/rust/ide/lib/span-tree/src/node.rs

```diff
@@ -138,8 +248,99 @@ pub struct Child {
     pub ast_crumbs : ast::Crumbs,
 }
 
+impl Deref for Child {
+    type Target = Node;
+    fn deref(&self) -> &Self::Target {
+        &self.node
+    }
+}
+
+impl DerefMut for Child {
+    fn deref_mut(&mut self) -> &mut Self::Target {
+        &mut self.node
+    }
+}
+
+
+
+// ====================
+// === ChildBuilder ===
+// ====================
+
+/// A builder pattern for `SpanTree`. A think wrapper for `Child` which adds useful methods for
+/// building properties of the current node.
+#[derive(Debug)]
+#[allow(missing_docs)]
+pub struct ChildBuilder {
+    pub child : Child,
+}
```

Comment:
We already have "ChildGenerator" in generate.rs, and more appriopriate way for building span-tree in `span-tree::builder::Builder`

Reply:
Add docs to both builders explaining why we keep them together

Reply:
I added docs. Moreover, I provided more explicit interface to the new builder, so it can be used for testing purposes as well! Lets clean this up when reworking the span tree API! :)